src/js/components/FileInput/FileInput.js

```diff
@@ -0,0 +1,246 @@
+import React, { forwardRef, useContext, useRef } from 'react';
+import styled, { ThemeContext } from 'styled-components';
+
+import { defaultProps } from '../../default-props';
+
+import { focusStyle, useForwardedRef } from '../../utils';
+
+import { Anchor } from '../Anchor';
+import { Box } from '../Box';
+import { Button } from '../Button';
+import { FormContext } from '../Form/FormContext';
+import { Keyboard } from '../Keyboard';
+import { Stack } from '../Stack';
+import { Text } from '../Text';
+
+import { StyledFileInput } from './StyledFileInput';
+
+const ControlBox = styled(Box)`
+  cursor: pointer;
+  &:focus {
+    ${focusStyle}
+  }
+  ${props => props.theme.fileInput && props.theme.fileInput.extend};
+  ${props =>
+    props.hover &&
+    props.theme.fileInput &&
+    props.theme.fileInput.hover &&
+    props.theme.fileInput.hover.extend};
+  ${props =>
+    props.dragOver &&
```

Comment:
Testing on the 'Multiple' story, the color from the `dragOver` theme maintains even after the cursor is not hovering.